lib/i18n/backend/active_record.rb

```diff
@@ -49,29 +49,28 @@ def store_translations(locale, data, options = {})
 
         def lookup(locale, key, scope = [], options = {})
           key = normalize_flat_keys(locale, key, scope, options[:separator])
-          result = Translation.locale(locale).lookup(key)
+          results = Translation.locale(locale).lookup(key).order(key: :desc)
 
-          if result.empty?
+          if results.empty?
             nil
-          elsif result.first.key == key
-            result.first.value
+          elsif results.first.key == key
+            results.first.value
           else
-            result = result.inject({}) do |hash, translation|
+            results.inject({}) do |hash, translation|
               hash.deep_merge build_translation_hash_by_key(key, translation)
-            end
-            result.deep_symbolize_keys
+            end.deep_symbolize_keys
           end
         end
 
         def build_translation_hash_by_key(lookup_key, translation)
-          hash = {}
           chop_range = (lookup_key.size + FLATTEN_SEPARATOR.size)..-1
-          translation_nested_keys = translation.key.slice(chop_range).split(FLATTEN_SEPARATOR)
-          translation_nested_keys.each.with_index.inject(hash) do |iterator, (key, index)|
-            iterator[key] = translation_nested_keys[index + 1] ?  {} : translation.value
-            iterator[key]
+          translation_nested_keys = translation.key.slice(chop_range)
+          return {} if translation_nested_keys.nil?
```

Comment:
Let's use `unless translation_nested_keys` instead of `if translation_nested_keys.nil?`